.circleci/detect-functional-changes.sh

```diff
@@ -2,12 +2,9 @@
 
 VERSION_CHANGE_TRIGGERS="setup.py MANIFEST.in openfisca_country_template"
 
-if [[ "$CIRCLE_BRANCH" == "master" ]]
-then LAST_MASTER_VERSION="HEAD^"
-else LAST_MASTER_VERSION="origin/master"
-fi
+last_tagged_commit=`git describe --tags --abbrev=0 --first-parent`  # --first-parent ensures we don't follow tags not published in master through an unlikely intermediary merge commit
```

Comment:
Not sure I fully understand the motive of the change.

Can you give a concrete example of a situation where the previous implem would not do its job?

One risk I can see here is if for some reason a tag is published in the middle of the feature branch. Then the script would not compare the feature branch to the last published version, but to this tag.

Reply:
This change has the technical benefit of working in both cases without needing specific logic based on the branch name.

In terms of logic, it makes the assumption that all tagged commits are published. The previous implementation made the assumption that all commits on `master` are published. I am more confident that no accidental tagging will happen than no commits on `master` can accidentally happen.
In the same way, if we want to unpublish a version (or mark a version as unpublished), we now simply have to remove a tag. Removing a commit on `master` is harder :wink:

If you want added rigidity (which would go beyond what was already the case), I'm happy to add a check for the tag to be `x.y.z`, preventing preversions from being taken into account 🙂

Reply:
Ok, fair enough.

I was thinking that under our workflow, accidental commits on master are pretty rare, if not inexistent, but we can't assume that it will be the same for all re-users.

Let's merge this as-is, and see if we need more refinement later.

Reply:
> under our workflow, accidental commits on master are pretty rare

Totally agree! It just happens that I believe the semantics of “this version was published” are more closely aligned with “this version is tagged” than “this version was merged into `master`”. Thanks to all the automation we've made, this is luckily all aligned, but I felt this implementation was both clearer and shorter 😉